File: src/main/java/org/voegtle/weatherwidget/diagram/ShenzhenDiagramActivity.java
package org.voegtle.weatherwidget.diagram;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import org.voegtle.weatherwidget.R;

public class ShenzhenDiagramActivity extends DiagramActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    addDiagram(DiagramEnum.shenzhen_7days);
    addDiagram(DiagramEnum.shenzhen_30days);
    addDiagram(DiagramEnum.shenzhen_lastyear);
  }


  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    MenuInflater inflater = getMenuInflater();
    inflater.inflate(R.menu.shenzhen_diagram_activity_menu, menu);
    return super.onCreateOptionsMenu(menu);
  }


  @Override
  protected boolean onCustomItemSelected(MenuItem item) {
    switch (item.getItemId()) {
      case R.id.action_7_days:
        viewPager.setCurrentItem(0, true);
        return true;
      case R.id.action_30_days:
        viewPager.setCurrentItem(1, true);
        return true;
      case R.id.action_last_year:
        viewPager.setCurrentItem(2, true);
        return true;
    }
    return false;
  }
}

File: src/main/java/org/voegtle/weatherwidget/diagram/ShenzhenDiagramActivity.kt
package org.voegtle.weatherwidget.diagram

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import org.voegtle.weatherwidget.R

class ShenzhenDiagramActivity : DiagramActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    addDiagram(DiagramEnum.shenzhen_7days)
    addDiagram(DiagramEnum.shenzhen_30days)
    addDiagram(DiagramEnum.shenzhen_7days)
  }


  override fun onCreateOptionsMenu(menu: Menu): Boolean {
    val inflater = menuInflater
    inflater.inflate(R.menu.shenzhen_diagram_activity_menu, menu)
    return super.onCreateOptionsMenu(menu)
  }


  override fun onCustomItemSelected(item: MenuItem): Boolean {
    when (item.itemId) {
      R.id.action_7_days -> return updateViewPager(0)
      R.id.action_30_days -> return updateViewPager(1)
      R.id.action_last_year -> return updateViewPager(2)
    }
    return false
  }
}
